File: frontend/src/app/models/Funcionario.ts
export interface Funcionario {
  id: number;
  nome: string;
  salario: number;
  dataNascimento: string;  // Definir como string para compatibilidade com JSON
  dataAdmissao: string;    // Definir como string para compatibilidade com JSON
  dataDemissao: string;    // Definir como string para compatibilidade com JSON
  cargo: string;
  endereco: string;
  telefone: string;
  email: string;
  cpf: string;
  rg: string;
  situacao: string;
}

File: backend/src/main/java/com/sistemaOficina/backend/entity/Funcionario.java
package com.sistemaOficina.backend.entity;
import lombok.Data;

import java.time.LocalDate;

import lombok.AllArgsConstructor;

@Data
@AllArgsConstructor
public class Funcionario {
    
    public Funcionario(Long id) {
        this.id = id;
    }
    private Long id;
    private String nome;
    private String salario;
    private LocalDate dataNascimento;
    private LocalDate dataAdmissao;
    private LocalDate dataDemissao;
    private String cargo;
    private String endereco;
    private String telefone;
}
